From 830b5806b508f180b44845b54d7d977706de5984 Mon Sep 17 00:00:00 2001
Date: Fri, 22 Jan 2021 11:38:12 +0800
Subject: Backport of JDK-8138971

summary: G1CollectorPolicy::_rs_lengths_prediction is not initialized before use
LLT:
Bug url: https://bugs.openjdk.java.net/browse/JDK-8138971
---
 hotspot/src/share/vm/gc_implementation/g1/g1CollectorPolicy.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/hotspot/src/share/vm/gc_implementation/g1/g1CollectorPolicy.cpp b/hotspot/src/share/vm/gc_implementation/g1/g1CollectorPolicy.cpp
index 761b0e953..046f8a848 100644
--- a/hotspot/src/share/vm/gc_implementation/g1/g1CollectorPolicy.cpp
+++ b/hotspot/src/share/vm/gc_implementation/g1/g1CollectorPolicy.cpp
@@ -122,6 +122,7 @@ G1CollectorPolicy::G1CollectorPolicy() :
                                 new TruncatedSeq(NumPrevPausesForHeuristics)),
 
   _recent_avg_pause_time_ratio(0.0),
+  _rs_lengths_prediction(0),
 
   _initiate_conc_mark_if_possible(false),
   _during_initial_mark_pause(false),
-- 
2.19.0

